### PR TITLE
Add: mikrotik_routeros_routing_ospf_interface_print_terse

### PR DESCRIPTION
**Pull request by kwood92**
Add support for parsing 'routing ospf interface print terse'

### CODE REVIEW

File: ntc_templates/templates/mikrotik_routeros_routing_ospf_interface_print_terse.textfsm [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Value INTERFACE (\S+)
+Value DYNAMIC (D)
+Value PASSIVE (P|passive)
+Value COST (\d+)
+
+Start
+  ^\s*\d+\s*(${DYNAMIC}\s*)?(${PASSIVE}\s*)?(comment=.+?)?interface=${INTERFACE}\s*cost=${COST}\s*priority=\d+\s*authentication=\S+\s*authentication-key=(?:")?\S+(?:")?\s*authentication-key-id=\d+\s*network-type=\S+\s*instance-id=\d+\s*retransmit-interval=\d+s\s*transmit-delay=\d+s\s*hello-interval=\d+s\s*dead-interval=\d+s\s*use-bfd=\S+$$ -> Record
+  ^\s*\d+\s*(D\s*)?address=\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}%${INTERFACE}\s*area=\S+\s*state=(${PASSIVE}|\S+)\s*network-type=\S+\s*cost=${COST}\s*(priority=\d+\s*)?use-bfd=\S+\s*retransmit-interval=\d+s\s*transmit-delay=\d+s\s*hello-interval=\d+s\s*dead-interval=\d+s\s*$$ -> Record
```
Comment (mjbear) [inline ntc_templates/templates/mikrotik_routeros_routing_ospf_interface_print_terse.textfsm:8]: Do you want to capture the D for dynamic in the terse01 raw output?
If this helps, please click the button to accept my commit suggestion. Thank you.

```suggestion
  ^\s*\d+\s*(${DYNAMIC}\s*)?address=\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}%${INTERFACE}\s*area=\S+\s*state=(${PASSIVE}|\S+)\s*network-type=\S+\s*cost=${COST}\s*(priority=\d+\s*)?use-bfd=\S+\s*retransmit-interval=\d+s\s*transmit-delay=\d+s\s*hello-interval=\d+s\s*dead-interval=\d+s\s*$$ -> Record
```

File: tests/mikrotik_routeros/routing_ospf_interface_print_terse/routing_ospf_interface_print_terse01.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  - cost: "1400"
+    dynamic: ""
+    interface: "vlan11@vlan975@sfp-sfpplus1"
+    passive: ""
+  - cost: "10"
+    dynamic: ""
+    interface: "vlan10@vlan975@sfp-sfpplus1"
+    passive: ""
+  - cost: "1"
+    dynamic: ""
```
Comment (mjbear) [inline tests/mikrotik_routeros/routing_ospf_interface_print_terse/routing_ospf_interface_print_terse01.yml:12]: Goes along with the above template suggestion/question about capturing D for dynamic.
Thank you!

```suggestion
    dynamic: "D"
    interface: "vlan11@vlan975@sfp-sfpplus1"
    passive: ""
  - cost: "10"
    dynamic: "D"
    interface: "vlan10@vlan975@sfp-sfpplus1"
    passive: ""
  - cost: "1"
    dynamic: "D"
```

File: ntc_templates/templates/mikrotik_routeros_routing_ospf_interface_print_terse.textfsm (hunk 1 above)
Comment (kwood92) [reply ntc_templates/templates/mikrotik_routeros_routing_ospf_interface_print_terse.textfsm:8]: In RouterOS 7, all interfaces in the 'routing ospf interface print terse' output are marked with 'D' even though the interfaces are not actually dynamically added into OSPF (unlike RouterOS 6), hence why it's ignored.

Comment (mjbear) [reply ntc_templates/templates/mikrotik_routeros_routing_ospf_interface_print_terse.textfsm:8]: Thank you @kwood92 for the details.
I don't work with RouterOS much these days so I wasn't aware. :face_with_peeking_eye:

File: ntc_templates/templates/mikrotik_routeros_routing_ospf_interface_print_terse.textfsm [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+  ^\s*$$
+  ^. -> Error
```
Comment (jmcgill298) [inline ntc_templates/templates/mikrotik_routeros_routing_ospf_interface_print_terse.textfsm:10]:
```suggestion
  ^. -> Error

```

File: tests/mikrotik_routeros/routing_ospf_interface_print_terse/routing_ospf_interface_print_terse.raw [hunk 4]
```diff
@@ -0,0 +1,3 @@
+ 0  P interface=all cost=10 priority=1 authentication=none authentication-key=foo authentication-key-id=1 network-type=broadcast instance-id=0 retransmit-interval=5s transmit-delay=1s hello-interval=10s dead-interval=40s use-bfd=no
+ 1    comment=ospf comment interface=vlan906@bond1 cost=10 priority=1 authentication=none authentication-key="bar" authentication-key-id=1 network-type=broadcast instance-id=0 retransmit-interval=5s transmit-delay=1s hello-interval=10s dead-interval=40s use-bfd=no
+ 2 DP interface=lo0 cost=10 priority=1 authentication=none authentication-key="" authentication-key-id=1 network-type=broadcast instance-id=0 retransmit-interval=5s transmit-delay=1s hello-interval=10s dead-interval=40s use-bfd=no
```
Comment (jmcgill298) [inline tests/mikrotik_routeros/routing_ospf_interface_print_terse/routing_ospf_interface_print_terse.raw:3]:
```suggestion
 2 DP interface=lo0 cost=10 priority=1 authentication=none authentication-key="" authentication-key-id=1 network-type=broadcast instance-id=0 retransmit-interval=5s transmit-delay=1s hello-interval=10s dead-interval=40s use-bfd=no

```

File: tests/mikrotik_routeros/routing_ospf_interface_print_terse/routing_ospf_interface_print_terse01.raw [hunk 5]
```diff
@@ -0,0 +1,3 @@
+0 D address=1.1.1.1%vlan11@vlan975@sfp-sfpplus1 area=INET-v2 state=ptp network-type=ptp cost=1400 use-bfd=no retransmit-interval=5s transmit-delay=1s hello-interval=1s dead-interval=4s
+1 D address=1.1.1.1%vlan10@vlan975@sfp-sfpplus1 area=MGMT-v2 state=ptp network-type=ptp cost=10 use-bfd=no retransmit-interval=5s transmit-delay=1s hello-interval=10s dead-interval=40s
+2 D address=1.1.1.1%Lo0 area=MGMT-v2 state=passive network-type=broadcast cost=1 priority=128 use-bfd=no retransmit-interval=5s transmit-delay=1s hello-interval=10s dead-interval=40s
```
Comment (jmcgill298) [inline tests/mikrotik_routeros/routing_ospf_interface_print_terse/routing_ospf_interface_print_terse01.raw:3]:
```suggestion
2 D address=1.1.1.1%Lo0 area=MGMT-v2 state=passive network-type=broadcast cost=1 priority=128 use-bfd=no retransmit-interval=5s transmit-delay=1s hello-interval=10s dead-interval=40s

```

**Review by jmcgill298 — APPROVED** [review]

**Pull request merged**